pkgs/development/python-modules/e3-core/default.nix

```diff
@@ -0,0 +1,58 @@
+{ buildPythonPackage
+, fetchPypi
+, colorama
+, pyyaml
+, python-dateutil
+, requests
+, requests-toolbelt
+, tqdm
+, tomlkit
+, typeguard
+, pytest
+, pytest-socket
+, httpretty
+, stevedore
+, mock
+, coverage
+, psutil
+, distro
+
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "e3-core";
+  version = "22.3.1";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-VhXV7NUGtnjeGMVIgrXwcVOLNTHNINm5Zg09udkCAN0=";
+  };
+
+  patches = [ ./0001-use-distro-over-ld.patch ];
+
+  propagatedBuildInputs = [
+    colorama pyyaml python-dateutil requests requests-toolbelt tqdm stevedore
+
+    # See setup.py:24. These are added dynamically for use on Linux. Darwin has
+    # its own set of requirements.
+    psutil distro
+  ];
+
+  buildInputs = [
+    tomlkit typeguard pytest pytest-socket httpretty mock coverage
```

Comment:
Use `pytestCheckHook` and related testing components in a `checkInputs` section.